app/controllers/admin/facilities_controller.rb

```diff
@@ -183,9 +183,11 @@ def initialize_upload
 
   def parse_file
     @file_contents = read_xlsx_or_csv_file(@file)
-    facilities = Facility.parse_facilities_from_file(@file_contents)
-    @errors = Csv::FacilitiesValidator.validate(facilities).errors
-    @facilities = facilities.map { |facility| facility.attributes.with_indifferent_access }
+    facility_data = Facility.parse_facilities_from_file(@file_contents)
+    @errors = Csv::FacilitiesValidator.validate(facility_data).errors
+    @facility_data_map = facility_data.transform_values! do |records|
```

Comment:
Yup, it's a nested hash. Structurally it looks like this:
```ruby
facility_data_map = {
    facilities: [
        # list of Facility models
    ],
    business_identifiers: [
        # list of FacilityBusinessIdentifier models
    ]
}
```

So the transform would turn all the model instances into plain old data maps–we need to do this because then it can be serialized when we perform the ActiveJob.

The other way I could have represented this is by combining the facilities and business identifiers and just have a single array, ie, like:
```ruby
facility_data_records = [
    {
        facility: # Facility model
        business_identifiers: [
            # list of FacilityBusinessIdentifier models, if any
        ]
    }, 
    # ...
]
```

I don't see an immediate benefit to clumping these models into a single representation. It adds some complexity to the transformations, and also adds some coupling (ie, make sure that the business identifier is clumped to the correct facility etc). Since our main purpose is to just pass these around and destructure in the Job, I kept it as straightforward as possible for now.

Reply:
Noted. I'll change it. I wasn't too sure about this decision, this (in)validation helped.

The model in my head was that we are uploading two sets of resources in one go, rather than a resource + its dependents. The latter seems more intuitive here.